src/board.ts

```diff
@@ -1,4 +1,5 @@
 import Pawn from './pieces/pawn';
+import Knight from './pieces/knight'
 import { Field } from '../src/types'
 
```

Comment:
Możemy ten kod trochę zmodyfikować żeby był wydajniejszy
1. Zauważ że koń ma ściśle zdefiniowane możliwości  ruchu 
Dlatego możemy zdefiniować wszystkoemożliwości  dla niego 

const moves = [[2, 1], [1, 2], [-2, 1], [1, -2], [-2, -1], [2, -1], [-1, 2], [-1, -2]];
następnie tworzymy jakgo nowe pozycje jako
 let newX = 0;
        let newY = 0;
        const attackMoves = [];
2. Następnie tworzymy możliwości ruchu dla konia

  for (const subTab of moves) {
            newX = this.x + subTab[0];
            newY = this.y + subTab[1];
            if (newX <= 7 && newX >= 0 && newY <= 7 && newY >= 0) {
                if (gameArea[newX][newY]) {
                    if (gameArea[newX][newY].side !== this.side || gameArea[newX][newY].side == this.side)
                        attackMoves.push(`${newX},${newY}`);
                } else attackMoves.push(`${newX},${newY}`);
            }
        }
NA koniec zwracamy naszą tablicę ruchów   return attackMoves i to wszytsko  oplatamy funkcją findAttackingMoves

Następniw tworzymy funkcję z wszystkimi możliwymi  ruchami w której występuje też attackingMoves, bo ruch atakujący też tak jakby zawiera się w możliwych ruchcach dlatego  wywołujemy ja w findLegalMoves lub possibleMoves nazwiesz jak będziesz chciała

    findLegalMoves(gameArea:Field[][]):string {
        let attack = this.findAttackingMoves(gameArea);
        let legal = [];

        for (const move of attack) {
            const attackX = move[0];
            const attackY = move[2];
            if (!gameArea[attackX][attackY] || gameArea[attackX][attackY].side !== this.side) {
                legal.push(move);
            }
     
        }

        return legal;
    }